src/python_testing/TC_RVCOPSTATE_2_1.py

```diff
@@ -0,0 +1,145 @@
+
+    @async_test_body
+    async def test_TC_RVCOPSTATE_2_1(self):
+
+        asserts.assert_true('PIXIT_ENDPOINT' in self.matter_test_config.global_test_params,
+                            "PIXIT_ENDPOINT must be included on the command line in "
+                            "the --int-arg flag as PIXIT_ENDPOINT:<endpoint>")
+
+        self.endpoint = self.matter_test_config.global_test_params['PIXIT_ENDPOINT']
+
+        attributes = Clusters.RvcOperationalState.Attributes
+
+        self.print_step(1, "Commissioning, already done")
+
+        if self.check_pics("RVCOPSTATE.S.A0000"):
+            self.print_step(2, "Read PhaseList attribute")
+            phase_list = await self.read_mod_attribute_expect_success(endpoint=self.endpoint, attribute=attributes.PhaseList)
+
+            if phase_list == NullValue:
+                logging.info("PhaseList is null")
+            else:
+                logging.info("PhaseList: %s" % (phase_list))
+
+                phase_list_len = len(phase_list)
+
+                asserts.assert_less_equal(phase_list_len, 32, "PhaseList must have no more than 32 entries!")
+
+        if self.check_pics("RVCOPSTATE.S.A0001"):
+            self.print_step(3, "Read CurrentPhase attribute")
+            current_phase = await self.read_mod_attribute_expect_success(endpoint=self.endpoint, attribute=attributes.CurrentPhase)
+            logging.info("CurrentPhase: %s" % (current_phase))
+
+            if phase_list == NullValue:
+                asserts.assert_true(current_phase == NullValue, "CurrentPhase should be null")
+            else:
+                asserts.assert_true(0 <= current_phase and current_phase <= (phase_list_len - 1),
+                                    "CurrentPhase must be between 0 and (phase-list-size - 1)")
+
+        if self.check_pics("RVCOPSTATE.S.A0002"):
+            self.print_step(4, "Read CountdownTime attribute")
+            countdown_time = await self.read_mod_attribute_expect_success(endpoint=self.endpoint, attribute=attributes.CountdownTime)
+
+            logging.info("CountdownTime: %s" % (countdown_time))
+            if countdown_time is not NullValue:
+                asserts.assert_true(countdown_time >= 0 and countdown_time <= 259200, "CountdownTime must be between 0 and 259200")
+
+        if self.check_pics("RVCOPSTATE.S.A0003"):
+            self.print_step(5, "Read OperationalStateList attribute")
+            operational_state_list = await self.read_mod_attribute_expect_success(endpoint=self.endpoint, attribute=attributes.OperationalStateList)
+
+            logging.info("OperationalStateList: %s" % (operational_state_list))
+
+            state_ids = []
+            state_labels = []
+            for s in operational_state_list:
+                state_ids.append(s.operationalStateID)
+                state_labels.append(s.operationalStateLabel)
+
+            # Defined states
+            definedStates = {0x00: 'Stopped',
```

Comment:
Can get these values from the codegen, eg `Clusters.RvcOperationalState.Enums.OperationalStateEnum.kSeekingCharger`

Reply:
actually, I think you could even do

`id in Clusters.RvcOperationalState.Enums.OperationalStateEnum.__members__.values()` in the assert below but I don't know how legal that is and you'd have to do it for the base class enum too. Which might be ok. 

At least make the keys the enum member names